src/device/onLoad.js

```diff
@@ -79,6 +79,215 @@ realityEditor.device.onload = async function () {
+        function(_newValue) {
+            // console.log('user wants cloudConnection to be', newValue);
+        },
+        function(_newValue) {
+            // console.log('cloud url text was set to', newValue);
+        }
+    );
+    let toggleNewNetworkId = realityEditor.gui.settings.addToggleWithFrozenText('New Network ID', 'generate new network id for cloud connection', 'generateNewNetworkId',  '../../../svg/object.svg', false, 'unknown', function(_newValue) {
+        // console.log('user wants newNetworkId to be', newValue);
+    });
+    let toggleNewSecret = realityEditor.gui.settings.addToggleWithFrozenText('New Secret', 'generate new secret for cloud connection', 'generateNewSecret',  '../../../svg/object.svg', false, 'unknown', function(_newValue) {
+        // console.log('user wants newSecret to be', newValue);
+    });
+
+    let cachedSettings = {};
+    const localSettingsHost = `localhost:${realityEditor.device.environment.getLocalServerPort()}`;
+
+    function processNewSettings(settings) {
+        let anyChanged = false;
+        if (cachedSettings.isConnected !== settings.isConnected) {
+            toggleCloudUrl.onToggleCallback(settings.isConnected);
+            anyChanged = true;
+        }
+        if ((cachedSettings.serverUrl !== settings.serverUrl) ||
+            (cachedSettings.networkUUID !== settings.networkUUID) ||
+            (cachedSettings.networkSecret !== settings.networkSecret)) {
+            anyChanged = true;
+            toggleCloudUrl.onTextCallback(`https://${settings.serverUrl}/stable` +
+                `/n/${settings.networkUUID}` +
+                `/s/${settings.networkSecret}`);
+            toggleNewNetworkId.onTextCallback(settings.networkUUID);
+            toggleNewSecret.onTextCallback(settings.networkSecret);
+        }
+        cachedSettings = settings;
+        if (anyChanged) {
+            document.getElementById("settingsIframe").contentWindow.postMessage(JSON.stringify({
+                getSettings: realityEditor.gui.settings.generateGetSettingsJsonMessage(),
+                getMainDynamicSettings: realityEditor.gui.settings.generateDynamicSettingsJsonMessage(realityEditor.gui.settings.MenuPages.MAIN)
+            }), "*");
+        }
+    }
+
+    // If we're viewing this on localhost we can connect to and read settings
+    // from the local server
+    if (window.location.host.split(':')[0] === localSettingsHost.split(':')[0]) {
+        fetch(`${realityEditor.network.useHTTPS ? 'https' : 'http'}://${localSettingsHost}/hardwareInterface/edgeAgent/settings`).then(res => res.json()).then(settings => {
+            processNewSettings(settings);
+        });
+    }
+    // Update settings when changed
+    realityEditor.network.realtime.subscribeToInterfaceSettings('edgeAgent', settings => {
+        processNewSettings(settings);
+    });
+
```

Comment:
The above block is something that was merged from main and not something I have added. Let me know if this is meant to be included in the code or if it was removed.

Reply:
I think this may be the result of a merge conflict. I had recently refactored a lot of the code out of the onLoad.js onload function, and moved it to a new file called `src/gui/settings/setupSettingsMenu.js`. So I think your previous commit (prior to "merged main into feature") is closer to what we want to merge in this PR

Reply:
I just confirmed this with Dan and updated the PR by removing this block of code and making another commit.